stdlib/streams/src/main/ballerina/streams/snapshot.bal

```diff
@@ -0,0 +1,350 @@
+# + numberOfCharacters - A `int` indicating number of chars to read.
+# + return - A `string` of content or an `error`.
+function readCharacters(io:ReadableCharacterChannel rch, int numberOfCharacters) returns string|error {
+    return rch.read(numberOfCharacters);
+}
+
+# Function to read all characters as a string from an io:ReadableCharacterChannel.
+# + rch - A `ReadableCharacterChannel` instance.
+# + return - A `string` of content or an `error`.
+function readAllCharacters(io:ReadableCharacterChannel rch) returns string|error? {
+    int fixedSize = 128;
+    boolean isDone = false;
+    string result = "";
+    while (!isDone) {
+        var readResult = readCharacters(rch, fixedSize);
+        if (readResult is string) {
+            result = result + readResult;
+        } else {
+            if (<string>readResult.detail()["message"] == "io.EOF") {
+                isDone = true;
+            } else {
+                return readResult;
+            }
+        }
+    }
+    return result;
+}
+
+# Function to write a string to a file using an io:WritableCharacterChannel.
+# + wch - A `WritableCharacterChannel` instance.
+# + content - The `string` content to be written.
+# + startOffset - A `int` indicating start offset.
+# + return - A `int` indicating number of chars got written or an `error`.
+function writeCharacters(io:WritableCharacterChannel wch, string content, int startOffset) returns int|error? {
+    return wch.write(content, startOffset);
+}
+
+# Function to initialize an io:ReadableCharacterChannel.
+# + filePath - A `string` path to the file.
+# + encoding - A `string` indicating the encoding type.
+# + return - The `ReadableCharacterChannel` which got initilized.
+function createReadableCharacterChannel(string filePath, string encoding) returns io:ReadableCharacterChannel? {
+    io:ReadableCharacterChannel? rch = ();
+    io:ReadableByteChannel|error byteChannel = trap io:openReadableFile(filePath);
+    if (byteChannel is io:ReadableByteChannel) {
+        rch = untaint new io:ReadableCharacterChannel(byteChannel, encoding);
+    }
+    return rch;
+}
+
+# Function to initialize an io:WritableCharacterChannel.
+# + filePath - A `string` path to the file.
+# + encoding - A `string` indicating the encoding type.
+# + return - The `WritableCharacterChannel` which got initilized.
+function createWritableCharacterChannel(string filePath, string encoding) returns io:WritableCharacterChannel? {
+    io:WritableCharacterChannel? wch = ();
+    io:WritableByteChannel|error byteChannel = trap io:openWritableFile(filePath);
+    if (byteChannel is io:WritableByteChannel) {
+        wch = untaint new io:WritableCharacterChannel(byteChannel, encoding);
+    }
+    return wch;
+}
+
+# Function to close writable/readable character channels.
+# + c - Character channel to be closed.
+function closeCharChannel(any c) {
+    if (c is io:ReadableCharacterChannel) {
+        var err = c.close();
```

Comment:
shouldn't we do something with `err`? same with L129?

Reply:
Fixed in abd66be